### ISSUE TITLE
test flake: file_linux_test.go:387: Simple pod: Expected update, timeout instead

### ISSUE THREAD

**Issue by quinton-hoole** [kubernetes/kubernetes#33979]
Seen in https://github.com/kubernetes/kubernetes/pull/33866#issuecomment-251252401

**Comment by quinton-hoole** [comment]
Logs at:

https://console.cloud.google.com/storage/browser/kubernetes-jenkins/pr-logs/pull/33866/kubernetes-pull-test-unit-integration/47574

**Comment by yujuhong** [comment]
```
=== RUN   TestReadPodsFromFileInDirAdded
E1003 15:55:43.235262   14001 file_linux.go:107] can't process config file "/tmp/dir-test989073204/test_pod_config_0": /tmp/dir-test989073204/test_pod_config_0: read '', but couldn't parse as pod(Object 'Kind' is missing in 'null').
--- FAIL: TestReadPodsFromFileInDirAdded (5.01s)
	file_linux_test.go:387: Simple pod: Expected update, timeout instead
```

The test expects two updates caused by two separate inotify events: CREATE, and MODIFY
```go
                // expect an update by CREATE inotify event
                expectUpdate(t, ch, testCase)
                // expect an update by MODIFY inotify event
                expectUpdate(t, ch, testCase)
```
It's my understanding that kubelet may send out only one update:
1. test creates a file.
2. inotify send a CREATE event.
3. kubelet tries to read the file, but the content is empty. kubelet reports `Object 'Kind' is missing in 'null'` error and does *not* generate an update.
4. test writes the full pod manifest to the file.
5. inotify sends a MODIFY event.
6. kubelet reads the pod manifest and sends and update.

Without knowing *when* kubelet will try to read the file, perhaps it's fair to expect only one update? 

/cc @wu8685

**Comment by wu8685** [comment]
@yujuhong I made a mistake in test case. The CREATE inotify is triggered before writing pod manifest in the file. It is  why the first `expectedUpdate` doesn't work: it expects a not null update until timeout.

Thanks @MrHohn to fix this bug in #33899 . I am so sorry to make this happen. I will be more carefully.

**Comment by yujuhong** [comment]
@wu8685 no problem. Closing this issue since #33899 has already been merged.
